src/Components/Api.js

```
@@ -0,0 +1,131 @@
+export default class Api {
+  constructor(id) {
+    this._id = id;
+    this._baseUrl = "https://around-api.en.tripleten-services.com/v1/";
+    this._authorization = "4135af44-f1c9-452d-8222-e09e3e6f1c85";
+  }
+
+  loadUserInfo() {
+    return fetch(`${this._baseUrl}users/me`, {
+      headers: { authorization: `${this._authorization}` },
+    })
+      .then((res) => {
+        if (res.ok) {
```

According to the task you need to throw an error if the response is not `ok`

```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

The code of `ok` checking is repeated in **every** request of the project. To avoid it you need to make a special method `_checkResponse` that will check any response from the server if it’s ok or not.


```js
_checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can replace redundant repeating with the only line:


```js
 .then(this._checkResponse)

```

Please, pay attention that you need to pass only the **reference** for the method rather than the call. 

**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.